network/p2p/sampler.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p2p
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/sampler"
+)
+
+var _ Sampler = (*UniformSampler)(nil)
+
+// Sampler samples connected peers
+type Sampler interface {
+	// Sample returns a nodeID from nodeIDs
+	// invariant: nodeIDs must not be modified
+	Sample(ctx context.Context, nodeIDs []ids.NodeID) (ids.NodeID, bool)
```

Comment:
Signature was changed to only return a single node since the intended use case for this was to choose a node to send a request to if the caller didn't provide one (i.e `AppRequestAny`). This also gets rid of edge-cases like what we have to do if the sampler implementation gives us more than the old `limit` parameter's amount of nodes.